fs/xfstests.py

```diff
@@ -512,89 +503,35 @@ def tearDown(self):
                             process.system('losetup -d %s' % dev, shell=True,
                                            sudo=True, ignore_status=True)
 
+    def _create_fsimages(self, loop_size, i):
+        dd_count = int(loop_size.split('GiB')[0])
+        if self.use_dd:
+            process.run('dd if=/dev/zero of=%s/file-%s.img bs=1G count=%s'
+                        % (self.disk_mnt, i, dd_count), shell=True,
+                        sudo=True)
+        else:
+            process.run('fallocate -o 0 -l %s %s/file-%s.img' %
+                        (loop_size, self.disk_mnt, i), shell=True,
+                        sudo=True)
+
     def _create_loop_device(self, loop_size, mount=True):
         if mount:
             self.part = partition.Partition(
                 self.base_disk, mountpoint=self.disk_mnt)
             self.part.mount()
-        # Creating two loop devices
-        for i in range(2):
-            if self.use_dd:
-                dd_count = int(loop_size.split('GiB')[0])
-                process.run('dd if=/dev/zero of=%s/file-%s.img bs=1G count=%s'
-                            % (self.disk_mnt, i, dd_count), shell=True,
-                            sudo=True)
-            else:
-                process.run('fallocate -o 0 -l %s %s/file-%s.img' %
-                            (loop_size, self.disk_mnt, i), shell=True,
-                            sudo=True)
+
+        # remove any previous losetup images & mounts
+        process.system('umount %s %s' % (self.scratch_mnt, self.test_mnt),
+                       sudo=True, ignore_status=True)
+        process.run('losetup -D')
```

Comment:
why are you removing all loop devices in system attached which are out of context from this test.
as `losetup -D` ->`Detach all associated loop devices.`

Reply:
better will be to check what next loop devices are available and only be using those.

Reply:
Sometimes when the test fails... it never gets to run the tearDown.. 
Hence I wanted to keep losetup -D.  I mostly see this problem while development, so anyways let me drop this patch.. 
I will debug the tearDown problem later.